packages/binding-mqtt/src/mqtt-client.ts

```diff
@@ -37,24 +37,24 @@ export default class MqttClient implements ProtocolClient {
         this.scheme = "mqtt" + (secure ? "s" : "");
     }
 
-    private client: any = undefined;
+    private client: mqtt.MqttClient = undefined;
 
     public subscribeResource(
         form: MqttForm,
-        next: (value: any) => void,
-        error?: (error: any) => void,
+        next: (value: Content) => void,
+        error?: (error: Error) => void,
         complete?: () => void
     ): Promise<Subscription> {
         return new Promise<Subscription>((resolve, reject) => {
             // get MQTT-based metadata
-            let contentType = form.contentType;
-            let retain = form["mqtt:retain"]; // TODO: is this needed here?
-            let qos = form["mqtt:qos"]; // TODO: is this needed here?
-            let requestUri = url.parse(form["href"]);
-            let topic = requestUri.pathname.slice(1);
-            let brokerUri: String = "mqtt://" + requestUri.host;
-
-            if (this.client == undefined) {
+            const contentType = form.contentType;
+            // const retain = form["mqtt:retain"]; // TODO: is this needed here?
+            // const qos = form["mqtt:qos"]; // TODO: is this needed here?
```

Comment:
I would get rid of those, if we'll need to re-introduce them we can. We can always use git to keep track of old code.

Reply:
removed commented out code